kernel: threads: assign index no. to dynamically created threads

dcpleung opened the pull request:
Kernel threads created at build time have unique indexes to map them into various bitarrays. This patch extends these indexes to dynamically created threads where the associated kernel objects are
allocated at runtime. Also adds some tests to make sure permissions are set correctly.

dcpleung committed:
tests/kernel: kernel.queue.poll: fails if MAX_THREAD_BYTES > 2

The kernel.queue.poll test fails if CONFIG_MAX_THREAD_BYTES is larger
than 2, complaining about no memory for semaphore object. Turns out
the memory pool is not large enough. So make it a bit larger.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

tyrantlizard77 reviewed (commented):
Mostly LGTM, minor comments.

This PR introduces 3 new test cases. Can we consolidate them into a single case?

dcpleung committed:
tests: kernel: add tests for dynamic threads

This adds two test cases to create dynamic threads, and one test
case to make sure permissions are set correctly.

Origin: Original

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
kernel: threads: assign index no. to dynamically created threads

Kernel threads created at build time have unique indexes to map them
into various bitarrays. This patch extends these indexes to
dynamically created threads where the associated  kernel objects are
allocated at runtime.

Fixes: #9081

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
Updated the commits addressing comment from @andrewboie.
